### PR TITLE
ci(actions): pin git SHA to the same value as GH Actions when running e2e tests on CircleCI

### PR DESCRIPTION
**Pull request by jijiechen**
CircleCI failed to load downloaded artifacts if there are new commits pushed in the branch during an ongoing run.

This PR is to pin the SHA that CircleCI workflows run on.

It should fix failures like this: https://app.circleci.com/pipelines/gh/kumahq/kuma/26972/workflows/be3668af-9369-4cf2-81d4-39e61694606a/jobs/566681

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues
  - N/A
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS
  - Confirmed 
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s)
  - N/A
  - Don't forget `ci/` labels to run additional/fewer tests
    - Added
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)?
  - No need
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) 
  - No need

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -747,6 +747,9 @@ workflows:
         equal: ["", << pipeline.parameters.gh_action_build_artifact_name >>]
     jobs:
       - e2e_testing:
+          filters:
+            tags:
+              only: /.*/
```
Comment (lahabana) [inline .circleci/config.yml:752]: Why is this needed?

**Review by lahabana — APPROVED** [review]
LGTM!

File: .circleci/config.yml (hunk 1 above)
Comment (jijiechen) [reply .circleci/config.yml:752]: Because CircleCI by default will run on branches and not any tags. They treat SHAs the same way as tags. 
Refer to this thread https://discuss.circleci.com/t/v2-api-pass-git-sha-to-trigger-new-pipeline/39811
